spec/30-processing-model.md

```diff
@@ -0,0 +1,63 @@
+potentially forwarded. This description can be used as a reference when
+implementing a Trace Context compliant tracing system, middleware - like a proxy
+or messaging bus - or cloud service.
+
+## Processing Model for Working with Trace Context
+
+The processing model describes the behavior of a tracer which modifies and
+forwards Trace Context headers.
+
+1. The tracer checks an incoming request for a `traceparent` and a `tracestate`
+   header.
+2. If no `traceparent` header is received, the tracer creates a new `trace-id`
+   and `parent-id` representing the current request.
+3. If `traceparent` header is present, the tracer tries to parse the version of
+   the `traceparent` header.
+   - If the version cannot be parsed, the tracer creates a new
+     `traceparent` header and deletes `tracestate`.
+   - If the version number is higher than supported by the tracer, the
+     implementation uses the format defined in this specification to parse
+     `trace-id` and `parent-id`. The tracer will only parse `trace-flags` values
+     supported by the current version of this specification and ignore all other
+     values. If parsing fails, the tracing system creates a new `traceparent`
+     header and deletes the `tracestate`.
+4. If the tracer supports the version number, it validates `trace-id` and
+   `parent-id`.
+   - If either `trace-id`, `parent-id` or `trace-flags` are invalid, the tracer
+     creates a new `traceparent` header and deletes `tracestate`.
+5. The tracer validates the `tracestate` header. If the `tracestate` header
+   cannot be parsed the tracer deletes it.
+6. For each outgoing request the tracer performs the following steps:
+  - The tracing system MUST modify the `traceparent` header.
+    - **Update `parent-id`**. The value of property `parent-id` MUST be set to a
+       value representing the ID of the current operation.
+    - **Update `recorded`**. The value of `recorded` reflects the caller's
+      recording behavior. The value of the `recorded` flag of `trace-flags` MAY
+      be set to `1` if the trace data is likely to be recorded or to `0`
+      otherwise. Setting the flag is no guarantee that the trace will be
+      recorded but increases the likeliness of end-to-end recorded traces.
+  - The tracer MAY modify the `tracestate` header
+     - **Update key value**. The value of any key can be updated. Modified keys
+        MUST be moved to the beginning of the list.
+     - **Add new key-value pair**. The new key-value pair should be added into
+       the beginning of the list.
+     - **Delete the key-value pair**. Any key-value pair MAY be deleted. It is
+       highly discouraged to delete keys that weren't generated by the same
+       tracing system or platform. Deletion of an unknown key-value pair will
+       break correlation in other systems. This mutation enables two scenarios:
+       - Proxies can block specific `tracestate` keys for privacy and security
```

Comment:
list broken now. Also, again - deleting of key-value pair has more applications than those two scenarios. If you want to explain that these two scenarios can justify the deletion of unknown keys - please tell this explicitly. Otherwise - let's delete those scenarios from here.

Reply:
This was your initial text. I can just delete the examples.

Reply:
Those are valid scenarios for the situation when you are deleting somebody else's tracestate pairs. This whole item about deleting in general, which has more valid use cases. My suggestion is to make it clear that this list of scenarios talking about exceptional cases, not the standard ones